src/coprocessor/statistics/cmsketch.rs

```diff
@@ -44,13 +46,32 @@ impl CmSketch {
         }
     }
 
+    pub fn sub(&mut self, bytes: &[u8], cnt: u32) {
+        self.count -= cnt;
+        let (h1, h2) = CmSketch::hash(bytes);
+        for (i, row) in self.table.iter_mut().enumerate() {
+            let j = (h1.wrapping_add(h2.wrapping_mul(i as u64)) % self.width as u64) as usize;
+            row[j] = row[j].saturating_sub(cnt);
+        }
+    }
+
+    pub fn push_to_top_n(&mut self, b: &[u8], cnt: u64) {
+        self.top_n.push((b.to_vec(), cnt))
+    }
+
     pub fn into_proto(self) -> tipb::CmSketch {
         let mut proto = tipb::CmSketch::default();
         let mut rows = vec![tipb::CmSketchRow::default(); self.depth];
         for (i, row) in self.table.iter().enumerate() {
             rows[i].set_counters(row.to_vec());
         }
         proto.set_rows(rows.into());
+        let mut top_n_data = vec![tipb::CmSketchTopN::default(); self.top_n.len()];
+        for (i, (data, cnt)) in self.top_n.iter().enumerate() {
```

Comment:
I suggest using `self.top_n.into_iter()` to save unnecessary allocation.

BTW, I prefer index-less style:
```rust
        for (pb_top_n_data, (data, cnt)) in top_n_data.iter_mut().zip(self.top_n) {
            pb_top_n_data.set_data(data);
            pb_top_n_data.set_count(cnt);
        }
```
Hope you like and adopt it.